code/obj/item/device/radios/intercoms.dm

```diff
@@ -13,6 +13,19 @@
 	rand_pos = 0
 	desc = "A wall-mounted radio intercom, used to communicate with the specified frequency. Usually turned off except during emergencies."
 	hardened = 0
+/obj/item/device/radio/intercom/set_dir(new_dir)
+	. = ..()
+	src.pixel_x = 0
+	src.pixel_y = 0
+	switch(src.dir)
+		if(NORTH)
+			pixel_y = -21
+		if(SOUTH)
+			pixel_y = 24
+		if(EAST)
+			pixel_x = -21
+		if(WEST)
+			pixel_x = 21
```

Comment:
Possible use for a component rather than adding behavior to `set_dir`? Search for `COMSIG_ATOM_DIR_CHANGED`. @pali6